bot/exts/valentines/valentine_zodiac.py

```diff
@@ -105,35 +107,33 @@ async def date_and_month(self, ctx: commands.Context, date: int, month: Union[in
             except ValueError as e:
                 final_embed = discord.Embed()
                 final_embed.color = Colours.soft_red
-                final_embed.description = f"Zodiac sign is not found because, {e}"
+                final_embed.description = f"Zodiac sign could not be found because.\n`{e}`"
                 log.info(f'Error in "zodiac date" command:\n{e}.')
             else:
                 final_embed = self.zodiac_build_embed(zodiac_sign_based_on_date)
 
         await ctx.send(embed=final_embed)
-        log.info("Zodiac sign embed based on date is now sent.")
+        log.trace("Embed from date successfully sent.")
 
     @zodiac.command(name="partnerzodiac", aliases=['partner'])
     async def partner_zodiac(self, ctx: commands.Context, zodiac_sign: str) -> None:
-        """Provides a counter compatible zodiac sign to the given user's zodiac sign."""
+        """Provides a random counter compatible zodiac sign to the given user's zodiac sign."""
         embed = discord.Embed()
         embed.color = Colours.pink
-        try:
-            compatible_zodiac = random.choice(self.zodiacs[zodiac_sign.lower()])
-            emoji1 = random.choice(HEART_EMOJIS)
-            emoji2 = random.choice(HEART_EMOJIS)
-            embed.title = "Zodiac Compatibility"
-            embed.description = (
-                f'{zodiac_sign.capitalize()}{emoji1}{compatible_zodiac["Zodiac"]}\n'
-                f'{emoji2}Compatibility meter : {compatible_zodiac["compatibility_score"]}{emoji2}'
-            )
-            embed.add_field(
-                name=f'A letter from Dr.Zodiac {LETTER_EMOJI}',
-                value=compatible_zodiac['description']
-            )
-        except KeyError:
-            embed = self.generate_invalidname_embed(zodiac_sign)
+        compatible_zodiac = random.choice(self.zodiacs[zodiac_sign.capitalize()])
```

Comment:
Won't this just throw now if you give an invalid zodiac sign?

Maybe assign the zodiacs to a variable using `self.zodiacs.get(zodiac_sign.capitalize())` and check if it's None before continueing instead of try/excepting.